src/coreclr/pal/src/misc/perfjitdump.cpp

```diff
@@ -61,13 +66,36 @@ namespace
         JIT_CODE_LOAD = 0,
     };
 
+    static bool UseArchTimeStamp()
+    {
+        static bool initialized = false;
+        static bool useArchTimestamp = false;
+
+        if (!initialized)
+        {
+#if defined(HOST_AMD64)
+            const char* archTimestamp = getenv("JITDUMP_USE_ARCH_TIMESTAMP");
```

Comment:
Instead of using getenv() directly you can add this to the other config settings the runtime supports:
1. Declare a config value in configvalues.h. Here is an example that uses the 'DontPrependPrefix' flag to declare an env var without the DOTNET_ prefix:
https://github.com/dotnet/runtime/blob/main/src/coreclr/inc/clrconfigvalues.h#L426

2. Use CLRConfig::GetConfigValue(CLRConfig::<symbol>) to retrieve the value. Example: https://github.com/dotnet/runtime/blob/cea112c04e144bb2ee6ee2515607fdc7dfb89d1a/src/coreclr/vm/profilinghelper.cpp#L634

Reply:
I think it would have to be `CLRConfigNoCache::Get("JITDUMP_USE_ARCH_TIMESTAMP", /*noprefix*/ true, &getenv)` to avoid layering issues with PAL. When this method is called with `noprefix=true`, it just calls `getenv` without anything extra.

Reply:
Thanks @jkotas! - I missed the pal layering aspect in there.

Reply:
Does this mean i dont have to update the configvalue.h and simply change the getenv method with what @jkotas said.

Reply:
I think calling `getenv` directly looks better.